engines/cpu.c

```diff
@@ -89,12 +263,27 @@ static int fio_cpuio_init(struct thread_data *td)
 
 	o->nr_files = o->open_files = 1;
 
-	log_info("%s: ioengine=%s, cpuload=%u, cpucycle=%u\n",
-		td->o.name, td->io_ops->name, co->cpuload, co->cpucycle);
+	switch (co->cpumode) {
+		case FIO_CPU_NOOP: noop_init(td);  break;
+		case FIO_CPU_QSORT: qsort_init(td); break;
+	}
 
+	// Let's restore the previous state and clear stats.
+	reset_all_stats(td);
+	td_set_runstate(td, td_previous_state);
 	return 0;
 }
 
+static void fio_cpuio_cleanup(struct thread_data *td)
+{
+	struct cpu_options *co = td->eo;
+
+	switch (co->cpumode) {
```

Comment:
Ditto on breaking it into multiple lines.

Reply:
yup fixed.